lang/cs/Org.Apache.REEF.IO/FileSystem/AzureDataLake/AzureDataLakeFileSystem.cs

```diff
@@ -0,0 +1,235 @@
+                throw new AdlsException($"Cannot create directory specified by {directoryUri.ToString()}");
+            }
+        }
+
+        /// <summary>
+        /// Deletes a directory.
+        /// </summary>
+        /// <exception cref="AdlsException">If directory cannot be deleted</exception>
+        public void DeleteDirectory(Uri directoryUri)
+        {
+            bool deleteStatus = Exists(directoryUri) && 
+                _adlsClient.GetDirectoryEntry(directoryUri.AbsolutePath).Type == DirectoryEntryType.DIRECTORY &&
+                _adlsClient.DeleteRecursive(directoryUri.AbsolutePath);
+            if (!deleteStatus)
+            {
+                throw new AdlsException($"Cannot delete directory specified by {directoryUri.ToString()}");
+            }
+        }
+
+        /// <summary>
+        /// Get the children on the given URI, if that refers to a directory.
+        /// </summary>
+        /// <exception cref="AdlsException">If directory does not exist</exception>
+        public IEnumerable<Uri> GetChildren(Uri directoryUri)
+        {
+            if (Exists(directoryUri) && _adlsClient.GetDirectoryEntry(directoryUri.AbsolutePath).Type == DirectoryEntryType.DIRECTORY)
+            {
+                foreach (var entry in _adlsClient.EnumerateDirectory(directoryUri.AbsolutePath))
+                {
+                    yield return new Uri($"{GetUriPrefix()}{entry.FullName}");
+                }
+            }
+            else
+            {
+                throw new AdlsException($"Cannot find directory specified by {directoryUri.ToString()}");
+            }
+        }
+
+        /// <summary>
+        /// Create Uri from a given file path.
+        /// The file path can be full with prefix or relative without prefix.
+        /// If null is passed as the path, ArgumentException will be thrown.
+        /// </summary>
+        /// <exception cref="ArgumentNullException">If specified path is null</exception>
+        public Uri CreateUriForPath(string path)
+        {
+            if (path == null)
+            {
+                throw new ArgumentNullException("Specified path is null");
+            }
+            return new Uri($"{GetUriPrefix()}/{path.TrimStart('/')}");
+        }
+
+        /// <summary>
+        /// Gets the FileStatus for remote file.
+        /// </summary>
+        /// <exception cref="ArgumentNullException">If remote file URI is null</exception>
+        /// <returns>FileStatus</returns>
+        public FileStatus GetFileStatus(Uri remoteFileUri)
+        {
+            if (remoteFileUri == null)
+            {
+                throw new ArgumentNullException("Specified uri is null");
+            }
+            var entrySummary = _adlsClient.GetDirectoryEntry(remoteFileUri.AbsolutePath);
+            if (!entrySummary.LastModifiedTime.HasValue)
+            {
+                throw new AdlsException("File/Directory at " + remoteFileUri + " does not have a last modified" +
```

Comment:
Can use string interpolation for `remoteFileUri `

Reply:
done